### DISCUSSION TITLE
Capture stdout of before_scenario and after_scenario

### DISCUSSION THREAD

**Discussion by ido-ran** [behave/behave#1058] [Q&A]
I was wondering if you'll consider having a way to capture the `print` and logging done in `before_scenario` and `after_scenario` with the test results captured log.
I am doing several things in those hooks and it will be very useful to be able to print and see the output in the test results captured in junit files.

I understand this is a behave change so it might require a specific settings to enable it.

What do you think?

**Comment by jenisys** [comment]
A hook decorator `capture()` for logging already exists in `behave.log_capture` module.

**Reply by jenisys** [reply]
Note that since `behave v1.3.0` (actually slightly before):

* Capture output was improved included to capture output of hooks

SEE ALSO:
* [features/capture.on_hooks.feature](https://github.com/behave/behave/blob/main/features/capture.on_hooks.feature)
* [features/capture.on_hooks.using_capture_output.feature](https://github.com/behave/behave/blob/main/features/capture.on_hooks.using_capture_output.feature)
